widget/ErrorInstancesTable.tsx

```diff
@@ -3,37 +3,101 @@
 
 import React, { Component } from "react";
 import ReactDOM from "react-dom";
-import * as d3 from "d3";
+import {
+  DetailsList,
+  DetailsListLayoutMode,
+  Selection,
+  SelectionMode,
+  IColumn
+} from "office-ui-fabric-react/lib/DetailsList";
+import { DefaultButton } from 'office-ui-fabric-react';
+import { Fabric } from "office-ui-fabric-react/lib/Fabric";
+import { apiBaseUrl } from "./api.ts";
 
 
 type ErrorInstancesTableState = {
-  data: any
+  selecedDataPoint: any,
+  page: number
 }
 
 type ErrorInstancesTableProps = {
-  data: any
+  selectedDataPoint: any,
+  pageSize?: number
 }
 
 class ErrorInstancesTable extends Component<ErrorInstancesTableProps, ErrorInstancesTableState> {
+
+  public static defaultProps = {
+      pageSize: 5
+  };
+
   constructor(props) {
     super(props);
-
     this.state = {
-      data: this.props.data
-    };
+      selecedDataPoint: null,
```

Comment:
Do we need to save selectedDataPoint in the state? It looks like we always read it from props

Reply:
Ah yes ... it is needed here because of Typescript type restrictions. Basically if you don't add this here the type checker complains that it is there in the type declaration and needs to be defined/set.

Reply:
Oh yeah and I just want the initial state to be a totally blank state. Which is why I'm setting this to null and pages to 0.
When we get stuff from props later it will get set.